File: 2001-3000/2149. Rearrange Array Elements by Sign.java
// Medium
// Two Pointers
// Time:O(n), Space:O(n)
// https://leetcode.cn/problems/rearrange-array-elements-by-sign/

class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] res = new int[n];
        int posIndex = 0, negIndex = 1;

        for (int x : nums) {
            if (x > 0) {
                res[posIndex] = x;
                posIndex += 2;
            } else {
                res[negIndex] = x;
                negIndex += 2;
            }
        }
        return res;
    }
}

File: 2001-3000/2149. Rearrange Array Elements by Sign.py
from typing import List

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos = [x for x in nums if x > 0]
        neg = [x for x in nums if x < 0]

        res = []
        n = len(nums) // 2
        for i in range(n):
            res.append(pos[i])
            res.append(neg[i])

        return res
